### PR TITLE
don’t reuse headers across HTTP calls

### LINKED ISSUE #9: Data race accessing header map
We're getting this output from Travis. It looks like there is a read to the header hashmap from lightstep while we're still editing it with backplane. I think you want to be taking a copy before unblocking.

```
WARNING: DATA RACE
Write by goroutine 9:
  runtime.mapassign1()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/hashmap.go:429 +0x0
  net/textproto.MIMEHeader.Set()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/textproto/header.go:22 +0xe1
  net/http.Header.Set()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/http/header.go:31 +0x56
  backplane/vendor/github.com/lightstep/lightstep-tracer-go/thrift_0_9_2/lib/go/thrift.(*THttpClient).Flush()
      /home/travis/gopath/src/backplane/vendor/github.com/lightstep/lightstep-tracer-go/thrift_0_9_2/lib/go/thrift/http_client.go:248 +0x2b1
  backplane/vendor/github.com/lightstep/lightstep-tracer-go/thrift_0_9_2/lib/go/thrift.(*TBinaryProtocol).Flush()
      /home/travis/gopath/src/backplane/vendor/github.com/lightstep/lightstep-tracer-go/thrift_0_9_2/lib/go/thrift/binary_protocol.go:456 +0x6a
  backplane/vendor/github.com/lightstep/lightstep-tracer-go/lightstep_thrift.(*ReportingServiceClient).sendReport()
      /home/travis/gopath/src/backplane/vendor/github.com/lightstep/lightstep-tracer-go/lightstep_thrift/reportingservice.go:81 +0x30a
  backplane/vendor/github.com/lightstep/lightstep-tracer-go/lightstep_thrift.(*ReportingServiceClient).Report()
      /home/travis/gopath/src/backplane/vendor/github.com/lightstep/lightstep-tracer-go/lightstep_thrift/reportingservice.go:55 +0x5d
  backplane/vendor/github.com/lightstep/lightstep-tracer-go.(*Recorder).Flush()
      /home/travis/gopath/src/backplane/vendor/github.com/lightstep/lightstep-tracer-go/recorder.go:324 +0x207a
  backplane/vendor/github.com/lightstep/lightstep-tracer-go.(*Recorder).reportLoop()
      /home/travis/gopath/src/backplane/vendor/github.com/lightstep/lightstep-tracer-go/recorder.go:441 +0x268
Previous read by goroutine 18:
  runtime.mapaccess1_faststr()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/hashmap_fast.go:193 +0x0
  net/textproto.MIMEHeader.Get()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/textproto/header.go:33 +0xa8
  net/http.Header.Get()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/http/header.go:39 +0x4e
  net/http.(*http2clientConnReadLoop).endStream()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/http/h2_bundle.go:5880 +0x227
  net/http.(*http2clientConnReadLoop).processData()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/http/h2_bundle.go:5863 +0x550
  net/http.(*http2clientConnReadLoop).run()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/http/h2_bundle.go:5607 +0x662
  net/http.(*http2ClientConn).readLoop()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/http/h2_bundle.go:5544 +0x20b
Goroutine 9 (running) created at:
  backplane/vendor/github.com/lightstep/lightstep-tracer-go.NewRecorder()
      /home/travis/gopath/src/backplane/vendor/github.com/lightstep/lightstep-tracer-go/recorder.go:208 +0x1015
  backplane/vendor/github.com/lightstep/lightstep-tracer-go.NewTracer()
      /home/travis/gopath/src/backplane/vendor/github.com/lightstep/lightstep-tracer-go/recorder.go:107 +0x129
  backplane/span.init.1()
      /home/travis/gopath/src/backplane/span/span.go:24 +0x1e5
  backplane/span.init()
      /home/travis/gopath/src/backplane/span/span.go:61 +0x105
  backplane/server.init()
      /home/travis/gopath/src/backplane/server/strategy.go:43 +0xbb
  backplane/database.init()
      /home/travis/gopath/src/backplane/database/database_test.go:1476 +0x9f
  main.init()
      backplane/database/_test/_testmain.go:100 +0x93
Goroutine 18 (running) created at:
  net/http.(*http2Transport).NewClientConn()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/http/h2_bundle.go:4975 +0x13e6
  net/http.(*http2addConnCall).run()
      /home/travis/.gimme/versions/go1.6.linux.amd64/src/net/http/h2_bundle.go:170 +0x6d
```

### PR DESCRIPTION
**Pull request by djspoons**
I believe this fixes #9

### CODE REVIEW

File: thrift_0_9_2/lib/go/thrift/http_client.go [hunk 1]
```diff
@@ -245,8 +245,16 @@ func (p *THttpClient) Flush() error {
 	p.requestBuffer = getBuffer()
 	p.responseBuffer = nil
 	// Do the call
-	p.header.Set("Content-Type", "application/x-thrift")
 	req.Header = p.header
+	// http2 reads the header outside of the call to Do(), so make a new
+	// header for next time here.
+	p.header = http.Header{}
```
Comment (bensigelman) [inline thrift_0_9_2/lib/go/thrift/http_client.go]: `p.header = make(http.Header, len(req.Header))`

File: thrift_0_9_2/lib/go/thrift/http_client.go [hunk 2]
```diff
@@ -245,8 +245,16 @@ func (p *THttpClient) Flush() error {
...
+	// Also, I'm not sure why the headers are reused from request to request,
```
Comment (bensigelman) [inline thrift_0_9_2/lib/go/thrift/http_client.go]: really looking forward to dethrifting.

**Comment by bensigelman** [comment]
LGTM aside from the nit about telling Go how big the map is going to be.

File: thrift_0_9_2/lib/go/thrift/http_client.go (hunk 1 above)
Comment (djspoons) [inline thrift_0_9_2/lib/go/thrift/http_client.go]: Yup, thanks. (Even though it's always empty for us. ;) )

**Pull request merged**